Учебный проект: консольные забавы

anatoliy-avramenko opened the pull request:
---
:mortar_board: [Учебный проект: консольные забавы](https://up.htmlacademy.ru/nodejs/3/user/457229/tasks/4)

Sergey-Rozhkov commented on readme.md:
```diff
@@ -1,7 +1,7 @@
 
 * Студент: [Anatoliy Avramenko](https://up.htmlacademy.ru/nodejs/3/user/457229).
-* Наставник: `Неизвестно`.
+* Наставник: [Максим Теплов](https://htmlacademy.ru/profile/id1031027).
```
неа, можешь потом поправить конечно

anatoliy-avramenko replied:
поправил

Sergey-Rozhkov commented on src/service/auxiliary/utils.js:
```diff
@@ -0,0 +1,21 @@
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+};
+
+const shuffle = (someArray) => {
+  for (let i = someArray.length - 1; i > 0; i--) {
+    const randomPosition = Math.floor(Math.random() * i);
+    [someArray[i], someArray[randomPosition]] = [someArray[randomPosition], someArray[i]];
+  }
+
+  return someArray;
+};
```
Способ конечно рабочий - но не красивый, как тебе такое ?
```
 const shuffle = (someArray) => someArray.sort(() => Math.random() - 0.5);
```

anatoliy-avramenko replied:
отличное предложение, спасибо! поправил

Sergey-Rozhkov commented on src/service/cli/generate.js:
```diff
@@ -0,0 +1,64 @@
+const getPictureFileName = (code) => {
+  const base = `item`;
+  const extension = `jpg`;
+  const processedCode = String(code).length === 1 ? `0${code}` : code;
```
```suggestion
  const processedCode = code.padStart(2, '0');
```

anatoliy-avramenko replied:
круто, не слышал про padStart() раньше. поправил

Sergey-Rozhkov commented on src/service/cli/generate.js:
```diff
@@ -0,0 +1,64 @@
+const fs = require(`fs`);
+
+const getPictureFileName = (code) => {
+  const base = `item`;
+  const extension = `jpg`;
```
мне кажется лишнее тут заведение константы

Sergey-Rozhkov commented on src/service/cli/generate.js:
```diff
@@ -0,0 +1,64 @@
+  const extension = `jpg`;
+  const processedCode = String(code).length === 1 ? `0${code}` : code;
+
+  return `${base}${processedCode}.${extension}`;
```
```suggestion
  return `item${code.padStart(2, '0')}.jpg`;
```

anatoliy-avramenko replied:
поправил, но `code` - это таки `number`, так что `String(code)`

Commit:
module1-task3 Добавление мелких исправлений
 - рефакторинг метода shuffle
 - рефакторинг метода getPictureFileName
 - исправление имени наставника

anatoliy-avramenko committed:
Merge branch 'master' into module1-task3
